partiql-lang/src/main/kotlin/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -150,6 +150,15 @@ internal open class EvaluatingCompiler(
             "compilationContextStack was empty.", ErrorCode.EVALUATOR_UNEXPECTED_VALUE, internal = true
         )
 
+    private val interruptionCheck: () -> Unit = when (compileOptions.interruptible) {
```

Comment:
Could you please add a comment here stating out where we currently conduct the interruptionCheck?